pkg/commands/volume.go

```diff
@@ -48,15 +48,15 @@ func (v *VolumeCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.
 	for _, volume := range resolvedVolumes {
 		var x struct{}
 		existingVolumes[volume] = x
-		err := util.AddVolumePathToWhitelist(volume)
+		err := util.AddPathToVolumeWhitelist(volume)
 		if err != nil {
 			return err
 		}
 
 		// Only create and snapshot the dir if it didn't exist already
 		if _, err := os.Stat(volume); os.IsNotExist(err) {
```

Comment:
[Severity-high]: By setting `v.snapshotFiles` to a single-element slice on each iteration, are we unintentionally discarding volumes from previous iterations? This might lead to only the last volume being considered.
Consider appending to `v.snapshotFiles` instead of overwriting it in each iteration. For example:
```suggestion
v.snapshotFiles = append(v.snapshotFiles, volume)
```